skimage/graph/_mcp.pyx

```diff
@@ -299,7 +299,12 @@ cdef class MCP:
         # but this is my code and I like fortran-style! Also, it's
         # faster when working with image arrays, which are often
         # already fortran-strided.)
-        self.flat_costs = costs.astype(FLOAT_D).flatten('F')
+        try:
+            self.flat_costs = costs.astype(FLOAT_D, copy=False).ravel('F')
+        except TypeError, ex:
```

Comment:
```
except Exception as ex
```

is what you want for testing purposes.  And then you can narrow it down a bit more.